dropwizard-client/src/test/java/io/dropwizard/client/DropwizardApacheConnectorTest.java

```diff
@@ -0,0 +1,180 @@
+package io.dropwizard.client;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.io.Resources;
+import io.dropwizard.Application;
+import io.dropwizard.Configuration;
+import io.dropwizard.jackson.Jackson;
+import io.dropwizard.setup.Environment;
+import io.dropwizard.testing.junit.DropwizardAppRule;
+import io.dropwizard.util.Duration;
+import org.apache.http.HttpStatus;
+import org.apache.http.conn.ConnectTimeoutException;
+import org.assertj.core.api.AbstractLongAssert;
+import org.glassfish.jersey.client.ClientProperties;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import java.net.SocketTimeoutException;
+import java.net.URI;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.CoreMatchers.any;
+
+public class DropwizardApacheConnectorTest {
+    private static final int SLEEP_TIME_IN_MILLIS = 500;
+    private static final int DEFAULT_CONNECT_TIMEOUT = 200;
+    private static final String NON_ROUTABLE_IP_ADDRESS = "10.255.255.1";
```

Comment:
So this is interesting. We found this IP on [stackoverflow](http://stackoverflow.com/a/904609) and it makes the test work and CI to pass. On the other hand, when I use the one you suggest I get 'SocketTimeoutException` rather than `ConnectionTimeoutException`. Even when I change the expected exception the test does not pass because the timeout settings are not observed.

So not sure what you want to do here. Shall we just drop the test if we cannot make it work?